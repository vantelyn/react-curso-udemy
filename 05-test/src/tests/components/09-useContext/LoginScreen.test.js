import { mount } from "enzyme";
import { LoginScreen } from "../../../components/09-useContext/LoginScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Pruebas en el componente <LoginScreen />', () => { 

    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={{
            setUser
        }}>
            <LoginScreen />
        </UserContext.Provider>
    );

    test('debe mostrarse correctamente', () => { 
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe llamar el método setUser al hacer click en el botón', () => {
        const click = wrapper.find( 'button' ).prop( 'onClick' );
        click();
        
        expect( setUser ).toHaveBeenCalledWith({
            id: 123,
            name: 'Fernando'
        });
    });

});