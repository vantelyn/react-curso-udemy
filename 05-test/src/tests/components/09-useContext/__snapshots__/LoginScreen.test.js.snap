// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <LoginScreen /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
